File: character-roleplay-server/src/main/java/com/example/characterroleplayserver/model/MsgResponse.java
package com.example.characterroleplayserver.model;

import com.fasterxml.jackson.annotation.JsonInclude;
import lombok.Data;
import lombok.experimental.SuperBuilder;
import org.springframework.http.HttpStatus;

import java.time.LocalDateTime;
import java.util.Map;

import static com.fasterxml.jackson.annotation.JsonInclude.Include.NON_NULL;

@Data
@SuperBuilder
@JsonInclude(NON_NULL)
public class MsgResponse {
    protected LocalDateTime timeStamp;
    protected int statusCode;
    protected HttpStatus status;
    protected String reason;
    protected String msg;
    protected String devMsg;
    protected Map<?, ?> data;
}

File: character-roleplay-website/src/app/interface/msg-response.ts
import { Character } from "./character";
import { User } from "./user";

export interface MsgResponse {
    timeStamp: Date;
    statusCode: number;
    status: string;
    reason: string;
    msg: string;
    devMsg: string;
    data: {characters?: Character[], character?: Character, users?: User[], user?: User}
} 
